src/combat/armor/armor-config.ts

```diff
@@ -0,0 +1,277 @@
+import { UUID } from 'java.util';
+import { Bukkit, Material } from 'org.bukkit';
+import { Attribute, AttributeModifier } from 'org.bukkit.attribute';
+import { Operation } from 'org.bukkit.attribute.AttributeModifier';
+import { EquipmentSlot, ItemStack } from 'org.bukkit.inventory';
+
+const ARMORS = new Map<Material, (item: ItemStack) => ItemStack>([
+  [Material.DIAMOND_HELMET, createDefaultHelmet],
+  [Material.DIAMOND_CHESTPLATE, createDefaultChestplate],
+  [Material.DIAMOND_LEGGINGS, createDefaultLeggings],
+  [Material.DIAMOND_BOOTS, createDefaultBoots],
+
+  [Material.LEATHER_HELMET, createDefaultHelmet],
+  [Material.LEATHER_CHESTPLATE, createDefaultChestplate],
+  [Material.LEATHER_LEGGINGS, createDefaultLeggings],
+  [Material.LEATHER_BOOTS, createCustomHat],
+
+  [Material.GOLDEN_HELMET, createDefaultHelmet],
+  [Material.GOLDEN_CHESTPLATE, createDefaultChestplate],
+  [Material.GOLDEN_LEGGINGS, createDefaultLeggings],
+  [Material.GOLDEN_BOOTS, createDefaultBoots],
+
+  [Material.CHAINMAIL_HELMET, createDefaultHelmet],
+  [Material.CHAINMAIL_CHESTPLATE, createDefaultChestplate],
+  [Material.CHAINMAIL_LEGGINGS, createDefaultLeggings],
+  [Material.CHAINMAIL_BOOTS, createDefaultBoots],
+
+  [Material.NETHERITE_HELMET, createDefaultHelmet],
+  [Material.NETHERITE_CHESTPLATE, createDefaultChestplate],
+  [Material.NETHERITE_LEGGINGS, createDefaultLeggings],
+  [Material.NETHERITE_BOOTS, createDefaultBoots],
+
+  [Material.IRON_HELMET, createIronHelmet],
+  [Material.IRON_CHESTPLATE, createIronChestplate],
+  [Material.IRON_LEGGINGS, createIronLeggings],
+  [Material.IRON_BOOTS, createIronBoots],
+
+  [Material.TURTLE_HELMET, createDefaultHelmet],
+]);
+
+/**
+ * Modify attributes of the armor
+ * @param item Armor to be modified.
+ * @returns Modified armor.
+ */
+export function modifyArmor(item: ItemStack) {
+  const modifierFunction = ARMORS.get(item.type);
+  if (!modifierFunction) {
+    return item;
+  }
+  return modifierFunction(item);
+}
+
+/**
+ * Default helmet with very small armor value
+ */
+function createDefaultHelmet(item: ItemStack) {
+  const meta = item.itemMeta;
+
+  const headAdd1 = new AttributeModifier(
```

Comment:
I think putting a modifier that is used exactly once to a variable is not necessary. If it is done, at the very least use descriptive variable name.

Same goes for all other cases of [slot]Add/SubX variables.